mz_compat.c

```diff
@@ -1111,12 +1118,17 @@ int unzLocateFile(unzFile file, const char *filename, unzFileNameComparer filena
         if (err != MZ_OK)
             break;
 
-        if ((intptr_t)filename_compare_func > 2) {
-            result = filename_compare_func(file, filename, file_info->filename);
-        } else {
-            int32_t case_sensitive = (int32_t)(intptr_t)filename_compare_func;
-            result = mz_path_compare_wc(filename, file_info->filename, !case_sensitive);
+        switch (filename_case) {
```

Comment:
Since `ignore_case` is now defaulted. this becomes:
```c
if (filename_case == 1) {
   ignore_case = 0;
} else if (filename_case > 1) {
   ignore_case = 1;
}
```

Reply:
And actually this logic can go outside of the `while` loop.

Reply:
Done.
Thanks again!